### DISCUSSION TITLE
We need a built in date Picker, could we get it into 5.2.0?

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#6020] [Ideas]
Folks,

The following is a non-plugin version of a date picker in wiki text

Try it on the prerelease, its makes use of the edit-text widget and needs a temporary location, then converts it to a tiddlywiki date serial number with a button click.
```
<$edit-text tiddler="$:/temp/birthday" tag="input" type=date/>
<$button tooltip="save selected date to bithday-date">
<$action-setfield $field="birthday-date" $value={{{ [{$:/temp/birthday}!is[blank]search-replace:g[-],[]addsuffix[120000000]] }}}/>
<$action-deletetiddler $tiddler="$:/temp/birthday"/>
Save Birthday
</$button>
```

Given this makes use of the html date picker and prerelease features for search and replace it is quite simple, but unnecessarily complex for everyday use. The above does not allow the date to be "seeded" with an existing value from the field content, and it seems unnecessary needing the submit button.

**I would like to suggest this be included such as in an $edit-date widget or core macro** so the core can support this fundamental need. We have a number of pickers, but not a date picker. Even if it does not have the bells and whistles this such a fundamental requirement, especially to ensure entered dates are valid.

My guess is this would be easy to implement.

Tony

**Comment by yaisog** [comment]
Couldn't you use the `inputActions` parameter of the EditTextWidget to at least avoid having to press the button? Works for me at least with the HTML date picker, but may need some extra persuation when trying to enter text manually (if the temporary tiddler is deleted in the inputActions call).

Yaisog
